src/ListenerCollectionFactory.php

```diff
@@ -22,54 +26,31 @@ public function __construct(Injector $injector, ContainerInterface $container)
 
     /**
      * @param array $eventListeners Event listener list in format ['eventName1' => [$listener1, $listener2, ...]]
-     * @param bool $precheck If true, all the listeners will be checked strictly if they are valid callables
-     *                       before attaching to the ListenerCollection
      *
      * @return ListenerCollection
      */
-    public function create(array $eventListeners, bool $precheck = false): ListenerCollection
+    public function create(array $eventListeners): ListenerCollection
     {
         $listenerCollection = new ListenerCollection();
 
         foreach ($eventListeners as $eventName => $listeners) {
-            if (!is_string($eventName)) {
+            if (!is_string($eventName) || !class_exists($eventName)) {
                 throw new InvalidEventConfigurationFormatException(
                     'Incorrect event listener format. Format with event name must be used.'
                 );
             }
 
-            if (!is_array($listeners)) {
-                $type = $this->isCallable($listeners) ? 'callable' : gettype($listeners);
+            if (!is_iterable($listeners)) {
+                $type = is_object($listeners) ? get_class($listeners) : gettype($listeners);
 
                 throw new InvalidEventConfigurationFormatException(
-                    "Event listeners for $eventName must be an array, $type given."
+                    "Event listeners for $eventName must be an iterable, $type given."
```

Comment:
Such check exists in `ListenerConfigurationChecker`. Should we duplicate it here?

Reply:
Yes, 4 lines after this one we do `foreach ($listeners...`. We just throw an exception instead of php error.